dockstore-integration-testing/src/test/java/io/dockstore/webservice/UserResourceIT.java

```diff
@@ -513,4 +513,41 @@ public void testUpdateUserMetadataFromGithub() {
         assertEquals("DockstoreTestUser2", userProfile.getUsername());
     }
 
+    @Test
+    public void testSetUserPrivilege() {
+        ApiClient adminWebClient = getWebClient(ADMIN_USERNAME, testingPostgres);
+        ApiClient userWebClient = getWebClient(USER_2_USERNAME, testingPostgres);
+
+        PrivilegeRequest privilegeRequest = new PrivilegeRequest();
+        UsersApi adminApi = new UsersApi(adminWebClient);
+        UsersApi userApi = new UsersApi(userWebClient);
+        User admin = adminApi.getUser();
+        User user = userApi.getUser();
+
+        privilegeRequest.setAdmin(false);
+        adminApi.setUserPrivilege(user.getId(), privilegeRequest);
+        assertFalse(userApi.getUser().isIsAdmin());
+        assertFalse(userApi.getUser().isCurator());
+
+        privilegeRequest.setCurator(true);
+        adminApi.setUserPrivilege(user.getId(), privilegeRequest);
+        assertTrue(userApi.getUser().isCurator());
+
+        try {
+            userApi.setUserPrivilege(admin.getId(), privilegeRequest);
+        } catch (ApiException ex) {
+            assertEquals(ex.getCode(), HttpStatus.SC_FORBIDDEN);
+        }
+
+        privilegeRequest.setAdmin(true);
+        adminApi.setUserPrivilege(user.getId(), privilegeRequest);
+        assertTrue(userApi.getUser().isIsAdmin());
+
+        privilegeRequest.setAdmin(false);
+        try {
+            adminApi.setUserPrivilege(admin.getId(), privilegeRequest);
+        } catch (ApiException ex) {
+            assertEquals(ex.getCode(), HttpStatus.SC_FORBIDDEN);
```

Comment:
Ditto

Reply:
Same for above, but in the case where the admin tries to remove themselves. Talked to Charles about it and it may be an issue if an admin were to accidentally remove admin privileges for themselves, which is why an error is thrown if an admin were to try and remove themselves.